pkg/codegen/go/gen_program.go

```diff
@@ -595,11 +600,22 @@ func GenerateProjectFiles(project workspace.Project, program *pcl.Program) (map[
 			version = "v" + p.Version.String()
 		}
 		if packageName != "" {
+			packagePaths[p.Name] = packageName
 			err = gomod.AddRequire(packageName, version)
 			contract.AssertNoErrorf(err, "could not add require statement for %s to go.mod", packageName)
 		}
 	}
 
+	// For any local dependencies, add a replace statement
+	for pkg, path := range localDependencies {
+		// pkg is the package name, we transformed these into Go paths above so use the map generated there
+		goPath, ok := packagePaths[pkg]
+		if ok {
+			err = gomod.AddReplace(goPath, "", path, "")
+			contract.AssertNoErrorf(err, "could not add replace statement to go.mod")
+		}
+	}
```

Comment:
Is this basically a bug fix we're now spotting before conformance tests actually exercise e.g. programgen better than the previous codegen tests?